### PR TITLE
fix wrong UX when adding attachments

### PR DESCRIPTION
**Pull request by lduboeuf**
this is a follow up from #172, closed unintentionally due to a mistake from my side
This avoid bad UX while opening the attachment popup while keyboard is shown and when removing an attachment
The previous PR was forgetting landscape mode and made it unusable when playing with attachments. Note that when having attachments in landscape, on small screen, you don't see the composing text area, it was already the case before.
-> EDIT: fixed with last commit

### CODE REVIEW

File: src/qml/ComposeBar.qml [hunk 1]
```diff
@@ -605,10 +614,12 @@ Item {
...
 
         onExpandedChanged: {
-            if (expanded && Qt.inputMethod.visible) {
-                attachmentPanel.forceActiveFocus()
+            //in landscape mode, we don't have enough place to display the attachment, lets dismiss keyboard
+            if (expanded && Qt.inputMethod.visible && (messages.landscape)) {
+                Qt.inputMethod.hide()
             }
         }
+
```
Comment (UniversalSuperBox) [inline src/qml/ComposeBar.qml]: This line of whitespace is not needed

File: src/qml/ComposeBar.qml [hunk 2]
```diff
@@ -605,10 +614,12 @@ Item {
         }
 
         onExpandedChanged: {
-            if (expanded && Qt.inputMethod.visible) {
-                attachmentPanel.forceActiveFocus()
+            //in landscape mode, we don't have enough place to display the attachment, lets dismiss keyboard
+            if (expanded && Qt.inputMethod.visible && (messages.landscape)) {
+                Qt.inputMethod.hide()
```
Comment (UniversalSuperBox) [inline src/qml/ComposeBar.qml]: Will this also cause the keyboard to dismiss on desktop, when the window is wider than it is tall?

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]

File: src/qml/ComposeBar.qml (hunk 2 above)
Comment (lduboeuf) [reply src/qml/ComposeBar.qml]: Humm, oh yes probably, that 's why there is attachmentPanel.forceActiveFocus() currently on xenial branch ?

**Comment by lduboeuf** [comment]
click test here: https://drive.google.com/file/d/1HRbyIjrOSOQXeZ00euIvV8twB4eOTXGq

**Comment by ghost** [comment]
Tested with FP2. I could attach a picture (and there was no bad UX with that), but then I unintentionally locked the screen. When I went back to my message, the picture was added an other time (there were two attachments of the same picture) ! However I could suppress them both.

**Comment by abkro** [comment]
Device: Nexus 5
Image: 2020-W35/2

Works nicely. To be more specific:
Tested in both portrait and landscape mode. Behavior described below.

Portrait mode: When selecting 'attachment' the keyboard remains and the options 'picture', 'video', and 'contact' appear. When one of them is selected, the keyboard disappears and one is presented with the different source options. Tapping the 'x' at the top left brings you back to the previous screen. Tapping the text entry field will then bring up the keyboard again.

Landscape mode: The functionality is identical, except that the keyboard disappears at the first step (when tapping the paperclip icon).

**Comment by jezek** [comment]
Device: FP2
Image: 2020-W35/2

Keyboard (and text) hiding in landscape when adding an attachment works.
The behavior is consistent with @abkro 's description.

I spotted an attachment (image) duplication, like @spartid described. It can be reproduced by:
- start message app in landscape mode.
- start composing new message (+ on top right, swipe from down, tapping on some contact conversation)
- if any attachments are already there (saved as draft), remove them.
- add a picture from gallery as attachment.
- rotate to landscape.
- add other picture from gallery. (the duplication happens if you choose not to add a picture from gallery and close he gallery with X on top right)

Result: There is a duplicate of first image in attachment.
Note: sometimes the duplication doesn't happen, but most times (7 of 9) it did.

**Comment by lduboeuf** [comment]
thanks for the report. Well i would say it is not a blocker for OTA-13 don't you think ?
We should file a bug

**Comment by jezek** [comment]
@lduboeuf I wouldn't know, if not mentioned above. I have rotation lock on and it newer occurred without screen rotation. For me, it's not a blocker. Filled issue #229

**Comment by ghost** [comment]
@lduboeuf I dont think it's a blocker for OTA-13, too. But I have rotation lock as well, and the duplication occurred there too. I'll mention it in issue #229

**Comment by Flohack74** [comment]
Alright not blocking the OTA.

**Pull request merged**